### PR TITLE
Default prepared_statements to false for mysql2 adapter

### PR DESCRIPTION
**Pull request by ilianah**
### Summary

This PR fixes a bug with defaulting `prepared_statements` to `true` for the mysql2 adapter prior to Rails 7.2 when this change should take effect (https://github.com/rails/rails/issues/42975)

As part of a [recent change](https://github.com/rails/rails/pull/44591), when initialising the mysql2 adapter [the abstract adapter constructor](https://github.com/rails/rails/blob/main/activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb#L45) is called first. The constructor [sets](https://github.com/rails/rails/blob/main/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb#L112) the value of `connection_parameters` and as part of this, the value of `prepared_statements`. In this case the default value for `prepared_statements` is `true` ([code](https://github.com/rails/rails/blob/main/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb#L127-L129)). 

Because the value of `connection_parameters` is set in the parent class, the [lines of code](https://github.com/rails/rails/blob/main/activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb#L59-L62) which used to set the default value to `false` for mysql2 adapter are no longer doing this.

Prior to this change, this value was set and passed down to the super constructor ([code](https://github.com/rails/rails/pull/44591/files#diff-538a9bf3450c3aed556a0d2eacc26839a60becd744cda8c0a247dcb06aa0f02fL59-L61))

### CODE REVIEW

File: activerecord/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+*  Disable `prepared_statements` for mysql2 adapter
```
Comment (eugeneius) [inline activerecord/CHANGELOG.md:1]:
```suggestion
*   Disable `prepared_statements` for mysql2 adapter
```

File: activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb [hunk 2]
```diff
@@ -171,6 +164,13 @@ def translate_exception(exception, message:, sql:, binds:)
             super
           end
         end
+
+        def default_prepared_statements # :nodoc:
```
Comment (eugeneius) [inline activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb:168]:
```suggestion
        def default_prepared_statements
```

This is a private method, so it's already undocumented.

File: activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb [hunk 3]
```diff
@@ -171,6 +164,13 @@ def translate_exception(exception, message:, sql:, binds:)
...
+          ActiveSupport::Deprecation.warn(<<-MSG.squish)
+          The default value of `prepared_statements` for the mysql2 adapter will be changed from +false+ to +true+ in Rails 7.2.
```
Comment (eugeneius) [inline activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb:170]:
```suggestion
            The default value of `prepared_statements` for the mysql2 adapter will be changed from +false+ to +true+ in Rails 7.2.
```

Just matching the indentation from the previous method.

File: activerecord/test/cases/adapters/mysql2/mysql2_adapter_test.rb [hunk 4]
```diff
@@ -71,6 +71,31 @@ def close
...
+      end
+
+      def query(*)
+      end
+
+      def close
+      end
+    end.new
+
+    adapter = assert_deprecated do
```
Comment (eugeneius) [inline activerecord/test/cases/adapters/mysql2/mysql2_adapter_test.rb:87]: I remembered the other method I was thinking of here:

```suggestion
    adapter = ActiveSupport::Deprecation.silence do
```

This won't fail if the deprecation is missing - that's the job of the test above this one.

File: activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb (hunk 2 above)
Comment (ilianah) [reply activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb:168]: Ah, yes, I removed it from the other method but forgot this one 🙈

**Pull request closed without merging**